apps/quick-learn-backend/src/routes/roadmap/roadmap.service.ts

```diff
@@ -52,53 +53,177 @@ export class RoadmapService extends BasicCrudService<RoadmapEntity> {
       .getMany();
   }
 
+  async findAllArchived(
+    user: UserEntity,
+    paginationDto: PaginationDto,
+  ): Promise<PaginatedResult<RoadmapEntity>> {
+    const queryBuilder = this.roadmapRepository
+      .createQueryBuilder('roadmap')
+      .leftJoinAndSelect('roadmap.roadmap_category', 'roadmap_category')
+      .leftJoinAndSelect('roadmap.created_by', 'created_by')
+      .leftJoinAndSelect('roadmap.updated_by', 'updated_by')
+      .leftJoinAndSelect('roadmap.courses', 'courses')
```

Comment:
Use only the left join as the course details will not be needed on the listing page.